changes/6137.fixed

```diff
@@ -0,0 +1 @@
+Fixed `related_field_name` on object tables in LocationType detail view.
```

Comment:
Is there a difference between `related_field_name` and `table_filter`? Can we combine them?

Reply:
I was wondering a bit about that myself. As I understand it:

- `related_field_name` is used as a URL query parameter both for the `add` button link (to pre-populate the related model form with a selection) and the "show all" link (as a filter parameter on the related model list view)
- `table_filter` is a queryset filter parameter used to get the correct queryset of related objects to render into the table.

In many (most?) cases they will be the same, but I have no doubt there will be exceptions at some point. Maybe it would make sense to default an unset `related_field_name` to fall through to the `table_filter` instead of defaulting to `obj._meta.model_name` though?

Reply:
I like that, not blocking for this PR. Feel free to merge